apps/dashboard/app/controllers/projects_controller.rb

```diff
@@ -65,7 +65,7 @@ def create
     else
       # TODO: loop through all errors and show them instead of this
       flash[:alert] = @project.errors[:name].last || @project.errors[:icon].last
-      redirect_to new_project_path(name: project_params[:name], icon: project_params[:icon])
+      redirect_to new_project_path(name: project_params[:name], directory: project_params[:directory], icon: project_params[:icon])
```

Comment:
Seems like we should only pass `project_params[:directory]` back only if it was defined. I initially ran into this error where it's empty (the url is `projects/new?directory=&icon=&name=3rd+default`) but the form is populated with the `data_root` which I then saved.

I'll file another ticket around verifiying that the directory you set is sane (i.e., is not the `data_root` itself)

You see here - this 4th project got created erroneously. 
```
[johrstrom dashboard(2558_added_implementation_for_project_path)]  cat data/projects/.project_lookup 
---
1: "/users/PZS0714/johrstrom/ondemand/src/apps/dashboard/data/projects/1"
2: "/users/PZS0714/johrstrom/ondemand/src/apps/dashboard/data/projects/2"
3: "/users/PZS0714/johrstrom/ondemand/src/apps/dashboard/data/projects/3"
4: "/users/PZS0714/johrstrom/ondemand/src/apps/dashboard/data/projects"
5: "/users/PZS0714/johrstrom/temp/test_project"
```

Reply:
I think the problem is with the project model and the population of fields done in the constructor and the save/update methods.

For this in particular, we could fix by adding a check for the id being populated:

```ruby
  #models/project.rb
  def initialize(attributes = {})
    if attributes.empty?
      @manifest = Manifest.new({})
    else
      @id = attributes.delete(:id)
      @directory = attributes.delete(:directory)
      @directory = Project.dataroot.join(id.to_s).to_s if @directory.blank? && @id

      @manifest = Manifest.new(attributes).merge(Manifest.load(manifest_path))
    end
  end
```

Reply:
On another note, why doing a redirect on failure?
Usually on HTTP post failures, the request is completed by rendering the form with the errors.

Reply:
regarding the error, maybe better to move the directory setup into the save method.

```ruby
  #models/project.rb
  def save(attributes)
    if id.nil? && attributes.fetch(:id, nil).nil?
      message = 'Project ID must be set when saving.'
      errors.add(:save, message)
      Rails.logger.warn(message)
      return false
    end

    @directory = Project.dataroot.join(id.to_s).to_s if @directory.blank?
    make_dir
    update(attributes)
  end
```

Reply:
Actually - we can defer the whole issue and fix it in #2976.